### ISSUE TITLE
In JS, @template with @typedef is not equivalent to a generic type alias

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#23745]
```js
/**
 * @template T
 * @typedef {[T, T]} TPair
 */
/** @type {TPair<number>} */
var tp = [1,2]
```

TPair doesn't correctly create a generic type alias.

**Expected behavior:**

TPair's typedef is equivalent to `type TPair<T> = [T, T]`

**Actual behavior:**

TPair's typedef is equivalent to `type T; type TPair = [T, T]` (where T is unique and inexpressible). So `@type {TPair}` is legal, but nothing can be assigned to it, and `@type {TPair<number>}` says that TPair is not generic.

**Comment by DanielRosenwasser** [comment]
Isn't this a duplicate of #23385?

**Comment by sandersn** [comment]
This is the first repro in #23385, which covered both template tags on classes and on typedefs. Notably, this bug has not come up in real code yet, only in our discussions and webpack’s discussions of style guidelines.

**Comment by sandersn** [comment]
Basically, there were two bugs to fix, so I went for the one that @TheLarkInn was waiting on. 😸

**Comment by TheLarkInn** [comment]
Thanks for submitting this, I wasn't 100% sure it was a bug but got distracted. I'm glad I was right so yay we can fix it! 😍😂

**Comment by sandersn** [comment]
Fix is up at #23947
